File: cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

float userinput();
int changeCount(int c);

int main(void)
{


 float dollars = userinput();

int cents = round(dollars * 100);
int r = changeCount(cents);

printf ("%i\n",r );
}







float userinput()
{
    float valueInput;
    do
    {
     valueInput = get_float("your change : ");

    }while(valueInput < 0);

 return valueInput;
}


int changeCount(int c){

int cc = c/25 + (c % 25)/10 + ((c % 25)%10)/5 + ((c % 25)%10)%5/1;

 return cc;
}

File: cash.py
# Wiritten in Python
 # Implement a program that calculates the minimum number of coins required to give a user change.
 # Assuming that the only coins available are quarters (25¢), dimes (10¢), nickels (5¢), and pennies (1¢).
 # Expected outcome form the command line:
    
 #  $ ./cash
 #  Change owed: -0.41
 #  Change owed: foo
 #  Change owed: 0.41
 #               4    

from cs50 import get_float

def main():

    ipt = get_input()
    cent = int((ipt*100))
    # output coin due
    print(f"Change owed :{changeCount(cent)}")


def get_input():

    while True:
        v_input = get_float("change owed : ")
        if v_input > 0:
            break
    return v_input


def changeCount(c):
    return (c//25 + (c % 25)//10 + ((c % 25) % 10)//5 + ((c % 25) % 10) % 5 // 1)


main()
